### PR TITLE
Fix issues with warp lenses on fabric

### PR DESCRIPTION
**Pull request by BenCat07**
This fixes two major issues with warp lenses on fabric, them destroying the force relays with bore, 

The below was intentional and removed from the pr. makes me sad though.

~~and the bursts no longer carrying any many after being warped.~~

Note that i auto formatted, so excuse me for formatting disrepancies, tell me how to fix them if desired.

### CODE REVIEW

File: src/main/java/vazkii/botania/common/item/lens/LensMine.java [hunk 1]
```diff
@@ -58,18 +61,17 @@ public boolean collideBurst(IManaBurst burst, HitResult rtr, boolean isManaBlock
...
+		if (!source.equals(collidePos) && !(tile instanceof IManaBlock) && neededHarvestLevel <= harvestLevel
+				&& hardness != -1 && hardness < 50F && (burst.isFake() || mana >= 24)) {
 			if (!burst.hasAlreadyCollidedAt(collidePos)) {
 				if (!burst.isFake()) {
 					List<ItemStack> items = Block.getDroppedStacks(state, (ServerWorld) world, collidePos, tile);
 
-					world.removeBlock(collidePos, false);
-					if (ConfigHandler.COMMON.blockBreakParticles.getValue()) {
-						world.syncWorldEvent(2001, collidePos, Block.getRawIdFromState(state));
+					if (!(world.getBlockState(collidePos).getBlock() instanceof BlockPistonRelay)) {
```
Comment (BenCat07) [inline src/main/java/vazkii/botania/common/item/lens/LensMine.java:70]: this is the actual change here

**Comment by BenCat07** [comment]
Was notified that i may not have fixed the actual issue, will look into more of it when i have time. Draft in the meantime.

**Comment by Hubry** [comment]
Remember to run Spotless next time because it failed your build on an unused import.

**Comment by williewillus** [comment]
can you check the forge version? why does the existing code work there and not here? That may indicate a deeper underlying bug. Papering it over at the surface level risks hiding that underlying issue.

Also, please run spotlessJavaApply on your branch

**Comment by BenCat07** [comment]
Thank you, will do the latter, checking the former later like i said

**Comment by BenCat07** [comment]
Turns out, combining the lenses in the wrong order was the cause, and it does the same thing on forge. i will see it as "intended" behaviour and close this pull request.

**Pull request closed without merging**